lib/transaction/index.js

```diff
@@ -393,23 +394,22 @@ transaction.prototype.requestAuth = function requestAuth(enrollment, options, ca
  *
  * @param {string} data.recoveryCode
  */
-transaction.prototype.recover = function recover(data) {
+transaction.prototype.recover = function recover(data, cb) {
   var self = this;
 
   if (!this.isEnrolled()) {
-    asyncHelpers.setImmediate(self.eventSequencer.emit.bind(self.eventSequencer), 'error',
-      new errors.NotEnrolledError());
+    asyncHelpers.setImmediate(events.emitErrorOrCallback, new errors.NotEnrolledError(), self, cb);
     return;
   }
 
   self.requestStrategyAuth(self.authRecoveryStrategy,
     function onRecoveryRequested(err, recoveryAuth) {
       if (err) {
-        self.eventSequencer.emit('error', err);
+        events.emitErrorOrCallback(err, self, cb);
```

Comment:
`events.emitErrorOrCallback` takes the emitter as second arg, but you send in self.  This would lead to corresponding `self.emit('error', err)`, not `self.eventSequencer.emit(...` as it was before.

Since `self` is inheriting EventEmitter.prototyp (I already regret having this oo logic hiding in the code base :-(  ), I guess this might work anyway, but every explicit emit in this file is done on the `self.eventSequencer`, not on `self`, so it is a bit confusing.